packages/activity/src/index.ts

```diff
@@ -283,33 +307,104 @@ export class Context {
    * :warning: Cancellation is not propagated from this function, use {@link cancelled} or {@link cancellationSignal} to
    * subscribe to cancellation notifications.
    */
-  public heartbeat = (details?: unknown): void => {
+  public readonly heartbeat = (details?: unknown): void => {
     this.heartbeatFn(details);
   };
 
-  /**
-   * Gets the context of the current Activity.
-   *
-   * Uses {@link https://nodejs.org/docs/latest-v14.x/api/async_hooks.html#async_hooks_class_asynclocalstorage | AsyncLocalStorage} under the hood to make it accessible in nested callbacks and promises.
-   */
-  public static current(): Context {
-    const store = asyncLocalStorage.getStore();
-    if (store === undefined) {
-      throw new Error('Activity context not initialized');
-    }
-    return store;
-  }
-
   /**
    * Helper function for sleeping in an Activity.
    * @param ms Sleep duration: number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
    * @returns A Promise that either resolves when `ms` is reached or rejects when the Activity is cancelled
    */
-  public sleep = (ms: Duration): Promise<void> => {
+  public readonly sleep = (ms: Duration): Promise<void> => {
     let handle: NodeJS.Timeout;
     const timer = new Promise<void>((resolve) => {
       handle = setTimeout(resolve, msToNumber(ms));
     });
     return Promise.race([this.cancelled.finally(() => clearTimeout(handle)), timer]);
   };
 }
+
+/**
+ * The current Activity's context.
+ */
+export function activityInfo(): Info {
+  // For consistency with workflow.workflowInfo(), we want activityInfo() to be a function, rather than a const object.
+  return Context.current().info;
+}
+
+/**
+ * The logger for this Activity.
+ *
+ * This is a shortcut for `Context.current().log` (see {@link Context.log}).
+ */
+export const log: Logger = new Proxy({} as Logger, {
```

Comment:
This doesn't have to be a proxy, you could hardcode all of the methods instead.

Reply:
Good point